ui/racer/src/view/race.ts

```diff
@@ -68,6 +77,12 @@ const renderTrack = (
   );
 };
 
+const hashCodeToVehicle = (str: string) => {
+  let h = 0;
+  for (let i = 0; i < str.length; i++) h = (Math.imul(31, h) + str.charCodeAt(i)) | 0;
+  return Math.abs(h) % 4;
+};
```

Comment:
where does this code come from?

Reply:
It was a code I found to transform a string into a number